src/lib/optimizer/chunk_statistics.cpp

```diff
@@ -0,0 +1,97 @@
+#include "chunk_statistics.hpp"
+
+#include <iterator>
+#include <sstream>
+
+#include "all_parameter_variant.hpp"
+#include "resolve_type.hpp"
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+namespace {
+template<typename T>
+std::shared_ptr<ChunkColumnStatistics> build_stats(const DictionaryColumn<T>& column) {
```

Comment:
This was a lazy hack because we were running into naming clashes with the function calling this (which is called `build_statistics`). Giving them very explicit names now.